compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -1170,16 +1171,20 @@ impl<'a> Parser<'a> {
         impl_dyn_multi: bool,
         ty: &Ty,
     ) {
+        #[derive(SessionDiagnostic)]
+        #[error(slug = "parser-maybe-report-ambiguous-plus")]
+        struct AmbiguousPlus {
+            pub sum_with_parens: String,
+            #[primary_span]
+            #[suggestion(code = "{sum_with_parens}")]
+            pub span: Span,
+        }
```

Comment:
Don't want to push more work on you, but this might be a good time to check if `multipart_suggestion` support works as expected.

Reply:
I'd be happy to do it but I couldn't find any examples on how to use it.